src/Peachpie.CodeAnalysis/CodeGen/VariableReference.cs

```diff
@@ -865,6 +865,20 @@ static void EmitTypeCheck(CodeGenerator cg, IPlace valueplace, SourceParameterSy
                     cg.EmitPop(cg.EmitCall(ILOpCode.Call, cg.CoreMethods.Operators.ThrowIfArgumentNull_object_int));
                 }
             }
+
+            // check callable
+            if (srcparam.Syntax.TypeHint.IsCallable(out bool isNullable))
```

Comment:
`isNullable` is not correct, do not use it. Parameter can also be `null` if its default value is `NULL`. Use `srcparam.IsNotNull` instead.
```php
function foo(callable $c = null) { } // <-- $c can be NULL
```